lib/internal/http2/core.js

```diff
@@ -3326,7 +3324,7 @@ function socketOnClose() {
 
     state.streams.forEach(closeStream);
     state.pendingStreams.forEach(closeStream);
-    session.close();
```

Comment:
`session.close()` calls  `session.destroy()` if `state.streams` and `state.pendingStreams` are empty. Therefore it could obstruct `session[kMaybeDestroy](err)`. Now it just marks the session as closed, hopefully fixing Windows: `test-http2-server-sessionerror.js` and `test-http2-too-many-settings.js` that do not create any streams.

Reply:
Indeed it fixed that!